### DISCUSSION TITLE
Server: some problems with the User ID number for people that connect

### DISCUSSION THREAD

**Discussion by computertech1012** [BearWare/TeamTalk5#1686] [Q&A]
Hi,
I am not sure if I should convert this into an issue or not, because I don't know how really to explain the problem.
So, basicly if you are connected to a server, let's say you are user ID number 4000 or something like that, eventually, if you reconnect, or if other people reconnect, eventually the ID number get set back to 1, or 2, and then continuing to go up after that.
This could cause a problem if you are using the ID feature for having MOTD variables in the server.
I will give you an example.
Let's say that my user name is "User" and I am user ID 3211. After a while several people connect and disconnect, and eventually the server goes back to resetting the ID number starting from like either 2 or something like that. I really don't know how to reproduce. But I think @bear101 would probably understand this. Let's say someone eventually connects with the user ID of 10. The person that connected with user ID number 10 would get the MOTD but it would show it as saying that last connected user was the user in number 3211. So if the MOTD said that the last connected user was %lastuser% then it would say last connected user was the user of 3211. I hope you understood what I am saying, it resets after a surten number, and I don't think it's 3211 that it resets, that was only by example.
Also note that if user 3211 were to disconnect, it would then take who ever was connnected with the higher number. So if someone is connected with say userid 3125, then it would for the next user say that the last user was that one instead of saying that the last user was the person that just connected.
If they were to disconnect and no numbers were higher than them it would resolve, but it only happens if the user count or whatever you call the User ID is reset back to lower numbers, like 1 or 2 or 3 or whatever it resets to. And that's what I am noticing.
I apologize, because I really don't know how to explain this. But eventually the server count ID number for the users is reset after the server has been running a while and people have reconnected and what not. That is what I am gathering from the server end anyway.
If you don't understand, or need me to explain a better way let me know and I will do my best.
Thanks for reading!

**Comment by bear101** [comment]
I don't understand the issue. %lastuser% is the user who has been connected for the least amount of time.

**Comment by lbk2907** [comment]
@bearware101
I think he ment the ID will reset to number 1 after reach 4000 plus plus. Maybe, but I not really sure.

**Reply by bear101** [reply]
Is that a problem?

**Comment by lbk2907** [comment]
For me is not a problem but I not sure how about him. 😊

**Reply by computertech1012** [reply]
Hi,
@lbk2907 If you look at a server that has a user id of someone with 4000, then anyone that connects to the server after that reset, as long as the id 4000 is still there, it will say welcome, and then the %lastuser% will display the one at 4000 rather than the user who connected. So instead of it saying welcome New User it will say Welcome Old User, where Old User is the ID with 4000. Maybe that's a better explination.
@bear101 Here is how to test. Make a server and make sure that the %lastuser% is configured and then after you do that, keep logging in until you know it's about to reset, say have a user ID at 4000. After that have another instance until it resets back to 1 o 2. After that reset, you will understand that when you log in as 2 and look at the MOTD it will say that 4000 was the last to log in when this is not true. Maybe now this is a better explination?
Thanks for reading!

**Reply by computertech1012** [reply]
Hi,
So, if you put let's say my user ID is 4000 and my name is John. The MOTD is configured to say the following.
"Welcome %lastuser%."
So instead of it saying welcome to the new user, that let's say has ID number 10, or restarts and goes to 11, and his name is Eric, then it will say Welcome John instead of Welcome Eric. And any other new user that connects will also see Welcome John instead of welcome and then their names. Is this a better explination? Maybe you can look at some of the servers in the public list, because it should say welcome stats but instead it says whatever that user is connected as, so if someone was connected with ID 4000 then the name won't change until they log out or that higher up ID is not present. Sorry I don't know how else to explain. I could make an audio recording and show you if you want.
Thanks for reading!

**Comment by Nardol** [comment]
An ID is only an identified, not something which has to be coherent with a statistic or a previous identifier.
It only has to be unique, not logical with the previous so I also don't understand why it is a problem.
In fact, that's why I personally don't find very useful to display an user ID as the last user who connected.
IMHO it would be a problem if there were two connected user with the same id.

**Comment by bear101** [comment]
it seems @computertech1012 is requesting a new variable like %currentuser% which is the user that just connected.

%lastuser% is the last user to connected, but not the current one.

**Reply by computertech1012** [reply]
Hi,
@bear101 This would work as well, however then why is it when the last user connects, it always says that the last user is the person that connects to that user, and doesn't say the user before him? For example say your user ID number 1 and your name is Joseph, and then I connect as Stats. It would say that Last connected user was Stats on my MOTD, while Joseph's says that it was Joseph. But yes, adding a new variable to this would also work.
But I think that it needs some work. Also, I think that if the user connects, it should update the MOTD for all clients at the top, and it should say the new last user that connected. Because if you go into server properties it does not change unless you log out or log in again. Same with %uptime%. It doesn't keep updating that while you are connected and it should. It just stays the same until you log out and back in again to get the new MOTD. It only updates if you actually physically update the MOTD.
But yes we could add a new variable, that would solve some of the problem.
Thanks for reading!

**Comment by computertech1012** [comment]
Hi,
Let me give some examples. For most of the public servers on the list it always is saying last connected user is Stats. This is an example, why is it not saying the user before Stats was connected? For example, I will paste some from my screen reader.
Unofficial server, Name: RTI Media Center, Users: 4, Country: SG, MOTD: Hi Stats!
Selamat datang di RTI Media Center!
Server ini di hosting oleh Gelegar Bangsa Multimedia.
Server ini telah aktif selama 0:06:58.
0 KBytes di terima dan 0 KBytes terkirim.
Saat ini ada 0 pengguna yang online dan 1 administrator yang sedang online di server.
Jika Stats membutukan bantuan, jangan ragu untuk menghubungi administrator yang sedang online di server.
Selamat menikmati. 

Unofficial server, Name: Gelegar Bangsa Multimedia TeamTalk Server, Users: 17, Country: SG, MOTD: Hi Stats!
Welcome to Gelegar Bangsa Multimedia TeamTalk Server.
This server has been active for: 23:46:50.
2853156 KBytes received and 1125976 KBytes sent.
There are currently: 17 users and 1 administrators online on the server.
I hope you feel comfortable and comfortable communicating and socializing here.
Best regards; GBM Teams. :)

Then here is one from the Green list. This is an example, every time someone connects it shows Ragu and he is user ID number 4056.

Public server, Name: Indonesian TeamTalk Server (ID), Users: 45, Country: ID, MOTD: Hai,
Raju L Miracle
Selamat datang di Indonesian TeamTalk Server!
Server ini didukung oleh: SMACI 
email: info @smaci.org.
Joind Telegram Group: http://karya.smaci.org/tgr!
Ada 44 pengguna terhubung ke server dan 2 administrator. Server ini telah aktif selama: 76:19:00
Jika membutuhkan sesuatu, jangan ragu untuk menghubungi administrator

Until that person disconnects, every single user that logs in will get that he is the last person, when that is false information. My user ID to connect there was 277. So since that other person was having a higher number this is false variable so maybe now you understood about it.
Thanks for reading!

**Comment by computertech1012** [comment]
Hi,
I wanted to take note that those server I published here used the %lastuser% variable but it actually shows the current user and that's the newest ID. So it should show the person before that user, but when no one else is online it should say no other user connected, or it should show another message because at this moment it is false.
